src/ape_test/accounts.py

```diff
@@ -10,41 +10,61 @@
 
 
 class TestAccountContainer(TestAccountContainerAPI):
-    _num_generated = 0
+    _num_generated: int
+    _accounts: List["TestAccount"]
+    _mnemonic: str
+    _num_of_accounts: int
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._num_generated = 0
+        self.init()
+
+    def init(self):
+        self._accounts = []
+        self._mnemonic = self.config["mnemonic"]
+        self._num_of_accounts = self.config["number_of_accounts"]
+        for index, account in enumerate(self._dev_accounts):
+            self._accounts.append(
+                TestAccount(
+                    index=index, address_str=account.address, private_key=account.private_key
+                )
+            )
+
+    def __len__(self) -> int:
+        return self._num_of_accounts
 
     @property
     def config(self):
         return self.config_manager.get_config("test")
 
     @property
     def _dev_accounts(self) -> List[GeneratedDevAccount]:
-        mnemonic = self.config["mnemonic"]
-        return generate_dev_accounts(mnemonic, number_of_accounts=self.config["number_of_accounts"])
+        return generate_dev_accounts(self._mnemonic, number_of_accounts=self._num_of_accounts)
 
     @property
     def aliases(self) -> Iterator[str]:
-        for index in range(len(self)):
+        for index in range(self._num_of_accounts):
             yield f"dev_{index}"
 
+    def _is_config_changed(self):
```

Comment:
`ProviderAPI`'s are allowed to implement a method called `update_settings()`.
We should use this method to change the mnemonic and handle regenerating accounts.
All you would do is add a method `update_settings()` to this class that takes a dictionary. And then check for mnemonic / num_of_accounts in there, set them, and do everything you do when you call `_is_config_changed()`

This way, you don't need to run `init` form within the `__len__` method and other stuff like that.
`update_settings()` exists for this reason!

Reply:
but, currently, temp_config context manager in ape is not using that just creating a temporary file as new config and using config.using_project to do the change, so update_settings wont be called anyway

Reply:
I cann change the context manager ofc but we may still get user errors i guess

Reply:
> but, currently, temp_config context manager in ape is not using that just creating a temporary file as new config and using config.using_project to do the change, so update_settings wont be called anyway

This was the reason i used this like this

Reply:
I suppose the issue is that you want to change the config while in the same Python session?

Reply:
yes, so that test wont fail, and i looked at the update_settings part and saw it is only for ProviderAPI and in here we dont need something like this